libwild/src/version_script.rs

```diff
@@ -90,63 +101,138 @@ impl<'data> MatchRules<'data> {
             }
         }
     }
+}
 
-    fn matches(&self, name: &PreHashed<UnversionedSymbolName>) -> bool {
-        if self.general.matches_all || self.general.exact.contains(name) {
-            return true;
-        }
+impl<'data> VersionScript<'data> {
+    fn find_match(
```

Comment:
@davidlattimore The current data structures and the lookup in this function is pretty chatty, do you have any tips&tricks how to make it nicer. A feedback would be appreciated :)